src/Npgsql/NpgsqlTypes/NpgsqlDate.cs

```diff
@@ -1,452 +1,235 @@
+        /// <summary>Gets the month component of the date represented by this instance.</summary>
+        /// <value>The month component, expressed as a value between 1 and 12.</value>
+        public int Month => NpgsqlDateTime.GetDatePart(Days, NpgsqlDateTimePart.Month);
 
-        #endregion
+        /// <summary>Gets the day component of the date represented by this instance.</summary>
+        /// <value>The day component, expressed as a value between 1 and 31.</value>
+        public int Day => NpgsqlDateTime.GetDatePart(Days, NpgsqlDateTimePart.Day);
 
-        #region Operators
+        /// <summary>Gets the number of days since PostgreSQL epoch.</summary>
+        /// <value>The number of days since PostgreSQL epoch.</value>
+        /// <seealso cref="PostgreSqlEpoch"/>
+        public int Days => _days;
 
-        public static bool operator ==(NpgsqlDate x, NpgsqlDate y) => x.Equals(y);
-        public static bool operator !=(NpgsqlDate x, NpgsqlDate y) => !(x == y);
-        public static bool operator <(NpgsqlDate x, NpgsqlDate y)  => x.CompareTo(y) < 0;
-        public static bool operator >(NpgsqlDate x, NpgsqlDate y)  => x.CompareTo(y) > 0;
-        public static bool operator <=(NpgsqlDate x, NpgsqlDate y) => x.CompareTo(y) <= 0;
-        public static bool operator >=(NpgsqlDate x, NpgsqlDate y) => x.CompareTo(y) >= 0;
+        /// <summary>
+        /// Determines whether the specified value is finite.
+        /// </summary>
+        /// <param name="value">A date.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified value is finite;
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        public static bool IsFinity(NpgsqlDate value) =>
+            value != NegativeInfinity &&
+            value != PositiveInfinity;
 
-        public static DateTime ToDateTime(NpgsqlDate date)
-        {
-            switch (date._type)
-            {
-            case InternalType.Infinity:
-            case InternalType.NegativeInfinity:
-                throw new InvalidCastException("Infinity values can't be cast to DateTime");
-            case InternalType.Finite:
-                try { return new DateTime(date._daysSinceEra * NpgsqlTimeSpan.TicksPerDay); }
-                catch { throw new InvalidCastException(); }
-            default:
-                throw new InvalidOperationException($"Internal Npgsql bug: unexpected value {date._type} of enum {nameof(NpgsqlDate)}.{nameof(InternalType)}. Please file a bug.");
-            }
-        }
+        /// <summary>
+        /// Determines whether the specified value is infinite.
+        /// </summary>
+        /// <param name="value">A date.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified value is infinite;
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        public static bool IsInfinity(NpgsqlDate value) =>
```

Comment:
As above, why not instance property

Reply:
Well, I designed it keeping in mind the existing design of BCL types, but I agree with that statics sucks. At the same time I would like to continue the trend made by Microsoft. If all here and @Brar agree that properties are better and avoiding Microsoft practices is a better option, I see no reason why we should stuck with static members.

Reply:
Which Microsoft practice are you referring to?

Reply:
Take a look at floating number types.

Reply:
There are definitely enough Microsoft examples where something has a property, rather than a static function (floats are also primitives, so not quite the same). But in any case I think we agree...

Reply:
FYI, it doesn't apply only to float numbers. It's even how `Rune` made which is quite new.